File: _deprecated/taewan/09.12/1291.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
  vector<int> answer;
  int low, high;
  void recursive(int num)
  {
    if (num > high)
      return;
    else if (num >= low && num <= high)
      answer.push_back(num);
    int last = num % 10;
    if (last == 9)
      return;
    int next = num * 10 + last + 1;
    recursive(next);
  }

  vector<int> sequentialDigits(int low, int high)
  {
    this->low = low, this->high = high;
    for (int i = 1; i < 10; i++)
      recursive(i);
    sort(answer.begin(), answer.end());
    return answer;
  }
};

int main(void) { return 0; }

File: sangjinHan/Study/8_1/1291.swift
//1291. Sequential Digits
/*
Example 1:

Input: low = 100, high = 300
Output: [123,234]
Example 2:

Input: low = 1000, high = 13000
Output: [1234,2345,3456,4567,5678,6789,12345]

367 783
456, 567, 678

1000000000
*/

let low = 1000
let high = 13000

func sequentialDigits(_ low: Int, _ high: Int) -> [Int] {
    var answer = [Int]()
	
	for i in 1..<10 {
		var num = i
		var next = i
		
		while (num <= high && next < 10) {
			if num >= low {
				answer.append(num)
			}
			
			next += 1
			num = num * 10 + next
		}
	}
	
	answer.sort()
	
	return answer
}


print(sequentialDigits(low,high))